### PR TITLE
server/api: api/init initialized

### LINKED ISSUE #1013: App init endpoint
Depends on #1014, #1012

add /api/init endpoint that returns {assessment, countries, region groups, regions }

### LINKED ISSUE #1028: move region_group to assessment schema
- [x] move tables region and region_group to _legacy schema
- [x] create table public.region 
region_code  text not null
        constraint region_pkey
            primary key,
    name         text
- [x] insert into  public.region (region_code, name)  values from _legacy.region
- [x] add table assessment_${name}.region_group
    id      integer not null
        constraint region_group_pkey
            primary key,
    name    varchar not null,
    "order" integer not null
- [x] rename assessment_${name}.assessment_country to assessment_${name}.country and 
assessment_${name}.assessment_region to assessment_${name}.region
- [x] to table  assessment_${name}.region add column region_group_id references assessment_${name}.region_group and make unique region_code + region_group_id

### PR DESCRIPTION
**Pull request by sorja**
Resolves #1013
Resolves #1028 ( https://github.com/openforis/fra-platform/pull/1024/commits/4f3549af719ab6327757f269e8f133848438b741 )

### CODE REVIEW

File: server/api/init/init.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      try {
+        const assessment = await AssessmentService.read({
+          assessment: {
+            props: {
+              name: assessmentName,
+            },
+          },
+        })
+
+        // Note: Use Schemas.getName(assessment)
```
Comment (sorja) [inline server/api/init/init.ts:24]: Note can be removed

File: server/repository/assessment_country/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { readAll } from './readAll'
+
+export const AssessmentCountryRepository = {
+  readAll,
+}
```
Comment (sorja) [inline server/repository/assessment_country/index.ts:5]: Use camelCase for folder names

File: server/service/regionGroup/readAll.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { BaseProtocol, DB } from '@server/db'
+import { RegionGroupRepository } from '@server/repository'
+import { RegionGroup } from '@core/meta/regionGroup'
+
+export const readAll = async (client: BaseProtocol = DB): Promise<RegionGroup[]> => {
```
Comment (sorja) [inline server/service/regionGroup/readAll.ts:5]: Prefer `Array<RegionGroup>`

File: server/service/assessmentRegion/readAll.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { BaseProtocol, DB } from '@server/db'
+import { AssessmentRegionRepository } from '@server/repository'
+import { Assessment } from '@core/meta/assessment'
+import { AssessmentRegion } from '@core/meta/assessmentRegion'
+
+export const readAll = async (
+  props: { assessment: Pick<Assessment, 'props'> },
+  client: BaseProtocol = DB
+): Promise<AssessmentRegion[]> => {
```
Comment (sorja) [inline server/service/assessmentRegion/readAll.ts:9]: Prefer `Array<AssessmentRegion>`

File: core/meta/assessmentCountry.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface AssessmentCountry {
```
Comment (minotogna) [inline core/meta/assessmentCountry.ts:1]: we dont need this

File: core/meta/assessmentRegion.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export interface AssessmentRegion {
```
Comment (minotogna) [inline core/meta/assessmentRegion.ts:1]: we dont need this

File: core/meta/cycle.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 export interface CycledPropsObject<Props = void> {
   props: {
-    cycles: Array<string>
+    cycles?: Array<string>
```
Comment (minotogna) [inline core/meta/cycle.ts:3]: this must be required

File: server/api/init/init.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  AssessmentService,
+  AssessmentCountryService,
+  AssessmentRegionService,
+  RegionGroupService,
+} from '@server/service'
+
+export const InitGet = {
+  init: (express: Express): void => {
+    express.get(ApiEndPoint.Init.one(), async (req: Request, res: Response) => {
+      const assessmentName = req.params.name ?? 'fra'
```
Comment (minotogna) [inline server/api/init/init.ts:14]: no fra hardcorded. read it from settings table if assessment name is not in the request

File: server/repository/assessmentCountry/index.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import { readAll } from './readAll'
+
+export const AssessmentCountryRepository = {
```
Comment (minotogna) [inline server/repository/assessmentCountry/index.ts:3]: we need AssessmentRepository.getCountries, AssessmentRepository.getRegions, AssessmentRepository.getRegionGroups,

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/api/init/init.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+          })
+        } else {
+          assessment = await AssessmentController.read({
+            name: assessmentName,
+          })
+        }
+
+        const countries = await AssessmentController.getCountries({ assessment })
+        const regions = await AssessmentController.getRegions({ assessment })
+        const regionGroups = await AssessmentController.getRegionGroups({ assessment })
```
Comment (minotogna) [inline server/api/init/init.ts:26]: you can use Promise.all to wait all 4 promises in parallel

File: server/controller/assessment/getRegions.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { BaseProtocol, DB } from '@server/db'
+import { AssessmentRepository } from '@server/repository'
+import { Assessment } from '@core/meta/assessment'
+
+export const getRegions = async (
+  props: { assessment: Pick<Assessment, 'props'> },
```
Comment (minotogna) [inline server/controller/assessment/getRegions.ts:6]: this can take just name in the props

File: server/controller/assessment/getCountries.ts [hunk 12]
```diff
@@ -2,11 +2,11 @@ import { BaseProtocol, DB } from '@server/db'
 import { AssessmentRepository } from '@server/repository'
 import { Assessment } from '@core/meta/assessment'
 
-export const read = async (
+export const getCountries = async (
   props: { assessment: Pick<Assessment, 'props'> },
```
Comment (minotogna) [inline server/controller/assessment/getCountries.ts:6]: this can take just name in the props

File: server/controller/assessment/getCountries.ts [hunk 13]
```diff
@@ -2,11 +2,11 @@ import { BaseProtocol, DB } from '@server/db'
...
   client: BaseProtocol = DB
-): Promise<Assessment> => {
+): Promise<Array<{ countryIso: string }>> => {
```
Comment (minotogna) [inline server/controller/assessment/getCountries.ts:8]: this should return an array of countries

File: server/controller/assessment/getRegions.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+  client: BaseProtocol = DB
+): Promise<
+  Array<{
+    regionCode: string
```
Comment (minotogna) [inline server/controller/assessment/getRegions.ts:10]: this should return an array of region

**Review by minotogna — CHANGES_REQUESTED** [review]

File: .github/workflows/test.js.yml [hunk 15]
```diff
@@ -36,6 +36,13 @@ jobs:
           PGDATABASE: frap-dev
           PGUSER: frap
           PGPASSWORD: frap
+      - run: yarn migrate-fra
```
Comment (minotogna) [inline .github/workflows/test.js.yml:39]: why do we have this in the tests? I dont think it should be here

File: server/controller/assessment/read.ts [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+import { Assessment } from '@core/meta/assessment'
+
+export const read = async (
+  props: { name: string } | { id: number },
+  client: BaseProtocol = DB
+): Promise<Assessment> => {
+  if ('id' in props) {
+    return AssessmentRepository.readById({ id: props.id }, client)
+  }
+  return AssessmentRepository.read({ name: props.name }, client)
```
Comment (minotogna) [inline server/controller/assessment/read.ts:12]: we can have just one method AssessmentRepository.read

File: server/repository/assessment/readById.ts [hunk 17]
```diff
@@ -0,0 +1,14 @@
+import { Assessment } from '@core/meta/assessment'
+import { BaseProtocol, DB } from '@server/db'
+import { Objects } from '@core/utils'
+
+export const readById = async (
```
Comment (minotogna) [inline server/repository/assessment/readById.ts:5]: we dont need this. add the conditional condition to AssessmentRepository.read

**Review by minotogna — CHANGES_REQUESTED** [review]

File: .github/workflows/test.js.yml (hunk 15 above)
Comment (sorja) [reply .github/workflows/test.js.yml:39]: For now, we use it to test against some data
Should we remove and remove tests that use data?

**Review by minotogna — APPROVED** [review]

**Pull request merged**